lib/BoundaryClickWatcher/index.js

```diff
@@ -0,0 +1,64 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+class BoundaryClickWatcher extends Component {
```

Comment:
This component is responsible for calling functions (passed down as props) when a user clicks outside of a component. To optimise events the outside click checking isn't done until the initial click is done and the component is active.